Document/0x05d-Testing-Data-Storage.md

```diff
@@ -414,6 +414,23 @@ public static int e(...);
 public static int wtf(...);
 }
 ```
+Please note that the above example only ensures the logging calls will be stripped. Any code for dynamic log statement generation, might be left in the bytecode, making the log statements accessible with a debugger <sup>[3]</sup>. According to the ProGuard author, latest version of ProGuard implement some heuristics to improve on this issue, however there is no guarantee, that complex string builders will be removed.
```

Comment:
> Any code for dynamic log statement generation, might be left in the bytecode, making the log statements accessible with a debugger.

I think the first comma should be removed.

> According to the ProGuard author, latest version of ProGuard implement some heuristics to improve on this issue, however there is no guarantee, that complex string builders will be removed.

This is a bit of a long sentence. I think it would be better if you split in two.

> only ensures the logging calls will be stripped. Any code for dynamic log statement generation, might be left in the bytecode.

After reading this it was not clear to me what the problem was. Maybe you can elaborate on it some more. Explain that it removes the call to the log function, but not the parameters to it, or something like that.